source/common/http/conn_manager_utility.cc

```diff
@@ -131,6 +131,63 @@ void ConnectionManagerUtility::mutateRequestHeaders(Http::HeaderMap& request_hea
   if (config.tracingConfig()) {
     Tracing::HttpTracerUtility::mutateHeaders(request_headers, runtime);
   }
+
+  if ((connection.ssl() && connection.ssl()->peerCertificatePresented()) ||
```

Comment:
nit: I still think it might be better to have this code be a separate routine, even if private. It's a lot to look through and the mutateRequestHeaders function is already huge. (If it was a separate routine I think it would probably be less nested because you could do early return). I could go either way on this but thought I would mention.